pkg/controllers/setup.go

```diff
@@ -166,12 +168,18 @@ func (s *Server) Start(ctx context.Context) error {
 		return fmt.Errorf("unable to setup readiness check: %v", err)
 	}
 
-	s.logger.Info("starting manager")
-	if err := mgr.Start(ctx); err != nil {
-		return fmt.Errorf("error starting manager: %v", err)
-	}
+	eg, egCtx := errgroup.WithContext(ctx)
+	eg.Go(func() error {
+		s.logger.Info("starting manager")
+		return mgr.Start(egCtx)
+	})
+
+	eg.Go(func() error {
+		s.logger.Info("starting webhook")
+		return webhook.SetupWebhookServer(egCtx, mgr, s.LeaderElectionNamespace)
+	})
```

Comment:
Question: any pros and cons of running webhook as a thread vs running as a separated container/pod?

Reply:
simplified packaging. No additional docker repos need to be setup etc.

plus the webhook is only going to serve a minimal change.